source/docs/magento1/add-custom-endpoint.md

```diff
@@ -0,0 +1,251 @@
+
+For the example with the social network, the directory is :
+
+- `[MODULE]/Model/Api2/Post/Rest/Guest/V1.php` (for guest mode)
+- `[MODULE]/Model/Api2/Post/Rest/Customer/V1.php` (for customer mode)
+
+This file is your API entrypoint.
+
+<blockquote class="warning">
+  **Warning:** Never forget to add Customer endpoint. If it is the same as the Guest endpoint, you can extend `Guest/V1.php` in `Customer/V1.php`. But if you don't do it, the logged in users won't be able to fetch data from your endpoint, breaking your feature once logged in.
+</blockquote>
+
+### Methods to implement
+
+- [GET] `protected function _retrieve()` call for `entity` action type
+- [GET] `protected function _retrieveCollection()` call for `collection` action type
+- [POST] `protected function _create()` call for `collection` action type, body is mandatory
+- [PUT] `protected function _update()` call for `entity` action type, body is mandatory
+- [PUT] `protected function _multiUpdate()` call for `collection` action type, body is mandatory
+- [DELETE] `protected function _delete()` call for `entity` action type, body is mandatory
+- [DELETE] `protected function _multiDelete()` call for `collection` action type, body is mandatory
+
+### Example
+
+```
+<?php
+
+/**
+ * Class Module_Network_Model_Api2_Post_Rest_Guest_V1
+ */
+class Module_Network_Model_Api2_Post_Rest_Guest_V1 extends FrontCommerce_Integration_Model_Api2_Abstract
+{
+    /**
+     * Get post list
+     *
+     * @return array
+     */
+    protected function _retrieveCollection()
+    {
+        $collection = $this->_getCollection();
+        $this->_applyCollectionModifiers($collection);
+        $this->_loadCollection($collection);
+        $this->addCacheHeaders($collection->getCacheLifetime());
+        $data = $collection->walk('toArray');
+        return array_values((array) $data);
+    }
+
+    /**
+     * Retrieve post collection
+     *
+     */
+    protected function _getCollection()
+    {
+        $collection = Mage::getResourceModel('module_network/post_collection');
+        return $collection;
+    }
+
+    /**
+     * Retrieve information about specified socialize
+     *
+     * @throws Mage_Api2_Exception
+     * @return array
+     */
+    protected function _retrieve()
+    {
+        $entityId = this->getRequest()->getParam('id'); // param name is defined on your route node, for this example is :id
+        /* @var $stockItem Mage_CatalogInventory_Model_Stock_Item */
+        $post = Mage::getModel('module_network/post')->load($entityId);
+        if (!$post || !$post->getId() || $post->getId() != $entityId) {
+            $this->_critical(self::RESOURCE_NOT_FOUND);
+        }
+        return $post->getData();
+    }
+}
+
+```
+
+### Testing
+
+In case you need to test your endpoints, you can either use cURL or [Postman](https://www.getpostman.com/). But advanced clients like Postman will allow you to have credentials and will make it easier to test customers vs guest endpoints.
```

Comment:
```suggestion
In case you need to ensure your endpoints works as expected, you can either use cURL or [Postman](https://www.getpostman.com/). But advanced clients like Postman will allow you to have credentials and will make it easier to run requests as logged in customers.
```